### DISCUSSION TITLE
GitHub Copilot failed to connect to server in WSL behind the proxy

### DISCUSSION THREAD

**Discussion by wenwei-lin** [community/community#63710] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

When I tried to install GitHub Copilot plugin in WSL VS Code, the plugin failed to run.

The Message showed that:

GitHub Copilot could not connect to server. Extension activation failed: "tunneling socket could not be established, cause=connect ECONNREFUSED 127.0.0.1:10809"

I have configured my WSL proxy as http://172.31.96.1:10809, but the extension fetch data through localhost.

Here is my configuration for wsl proxy:

![image](https://github.com/community/community/assets/75592450/6669bf0e-8015-488c-83d1-29dfd5928493)

My VS Code configuration is here:

![image](https://github.com/community/community/assets/75592450/00996e9e-2f42-4191-a7dc-f5daba7a2082)

My wsl can ping to https://copilot-proxy.githubusercontent.com/_ping successfully.

![image](https://github.com/community/community/assets/75592450/9c20b5da-b507-4f18-b1fe-42a431a51ba4)

### OS Info

- Edition	Windows 11 Pro Insider Preview
- Version	22H2
- Installed on	‎8/‎11/‎2023
- OS build	23521.1000
- Experience	Windows Feature Experience Pack 1000.23521.1000.0

**Comment by Akash1134** [comment]
Hi @wenwei-lin ,

I've shared your issue and report with our team internally. I'll get back to you when I hear back from them. Thanks! 🙂

**Comment by hsenag** [comment]
Can you explain your setup a bit more and why you need to setup a proxy at all - is it that you have some firewall and are running a proxy on the Windows side which is the only way to get https traffic out?

Our code should just pick up the proxy from the environment variable. I can't think of a reason it would be getting the right port but the wrong IP, given they both come from the same place.

**Reply by wenwei-lin** [reply]
My network is behind a firewall set by government. So I need a proxy to use some service.

I was wondering about the port too. How it can figure out my port.

After reinstall my WSL with latest distribution, I can use Copilot now.

**Comment by Jaber-Saed** [comment]
It looks like you're facing an issue with GitHub Copilot's connection within WSL (Windows Subsystem for Linux) in Visual Studio Code. The error message suggests that there's an issue establishing a connection through a proxy.

Here's what you can try to troubleshoot and resolve the issue:

    Proxy Configuration in WSL:
    Make sure that your WSL environment is using the correct proxy settings. You mentioned that you've configured your WSL proxy as http://172.31.96.1:10809, which seems to be an IP address and port for your proxy server. Double-check that these settings are accurate.

    Check Proxy Availability: Ensure that the proxy server you're using (172.31.96.1:10809) is up and running, and it's not blocking connections from WSL. You can try accessing websites from within your WSL terminal to confirm that the proxy settings are working as expected.

    Extension Environment Variables: Sometimes, extensions require environment variables to be set for proper functionality. Check if the GitHub Copilot extension requires any specific environment variables related to proxies or network settings. You might need to set these variables in your WSL environment. You can do this by modifying your .bashrc or .bash_profile file in your WSL home directory.

    VS Code Settings:
    In your WSL environment, open the VS Code settings (either from the menu or by editing the settings.json file). You can try explicitly specifying proxy settings within VS Code's settings to ensure that the extension uses the correct proxy:

    json

    "http.proxy": "http://172.31.96.1:10809",
    "https.proxy": "http://172.31.96.1:10809"

    Restart VS Code and WSL:
    After making changes to your proxy settings or VS Code settings, make sure to fully close and restart both VS Code and your WSL terminal to apply the changes.

    Check Firewall Settings:
    If you have a firewall or security software, make sure that it's not blocking the connections that GitHub Copilot needs to establish.

    GitHub Copilot Known Issues:
    It's worth checking if there are any known issues or solutions related to GitHub Copilot and proxy setups. GitHub's official documentation or forums might have insights into similar problems and solutions.

If you've tried these steps and the issue persists, you might consider reaching out to GitHub Copilot support or the Visual Studio Code community for further assistance. Sometimes, specific extensions may have unique requirements or conflicts that require expert guidance to resolve.

**Reply by wenwei-lin** [reply, marked as answer]
I reinstalled my WSL and it works now. I think the problem caused by Windows Insider Update.

**Comment by FabiJa** [comment]
For information: 
I had a similar problem with WSL and proxy leading to the above error message. My problem was that I had configured the proxy not only with the env variable `http_proxy`, but also in the VS Code settings. After removing the proxy config in the VS code settings (see screenshot above), Copilot works.
